src/main/java/co/rsk/federate/btcreleaseclient/BtcReleaseClient.java

```diff
@@ -252,21 +281,55 @@ public void onBlock(org.ethereum.core.Block block, List<TransactionReceipt> rece
             pegoutTxs.forEach(BtcReleaseClient.this::onBtcRelease);
         }
 
+        /**
+         * Determines if the svp spend transaction hash is ready to be signed based on its block confirmations.
+         *
+         * <p>
+         * This method retrieves the block associated with the given transaction hash and calculates
+         * the difference in block numbers between the current block and the block containing the transaction.
+         * If the difference meets or exceeds the required confirmation threshold defined in the bridge constants,
+         * the transaction is considered ready for signing.
+         * </p>
+         *
+         * @param currentBlockNumber the current block number in the blockchain
+         * @param svpTxHash the Keccak256 hash of the svp spend transaction waiting to be signed
+         * @return {@code true} if the transaction has the required number of confirmations and is ready to be signed;
+         *         {@code false} otherwise
+         */
+        private boolean isSVPSpendTxReadyToSign(long currentBlockNumber, Keccak256 svpTxHash) {
+            boolean isReadyToSign = Optional.ofNullable(svpTxHash)
+                .map(Keccak256::getBytes)
+                .flatMap(txHash -> receiptStore.getInMainChain(txHash, blockStore))
+                .map(TransactionInfo::getBlockHash)
+                .map(blockStore::getBlockByHash)
+                .map(Block::getNumber)
+                .map(blockNumberWithSvpSpendTx -> currentBlockNumber - blockNumberWithSvpSpendTx)
+                .filter(confirmationDifference -> confirmationDifference >= bridgeConstants.getRsk2BtcMinimumAcceptableConfirmations())
+                .isPresent();
+
+            logger.info("[isReadyToSign] SVP spend tx readiness check for signing: tx hash [{}], Current block [{}], Ready to sign? [{}]",
+                svpTxHash,
+                currentBlockNumber,
+                isReadyToSign ? "YES" : "NO");
```

Comment:
```suggestion
            logger.info("[isSvpSpendTxReadyToSign] SVP spend tx should be ready to be signed: tx hash [{}], Current block [{}]",
                svpTxHash,
                currentBlockNumber;
```

wdyt? Although after 4K confirmations the tx for sure will be ready to sign, i'll try to avoid **confirming** it here